usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js

```diff
@@ -532,42 +531,44 @@
         /* On releasing trigger, if the user is not in whiteboard zone, delete this sphere. If laser or drawing interval 
         exists, delete it.*/
         triggerReleased: function() {
-            if (!_this.isUserInZone(whiteboardZone)) {
+            if (!_this.isUserInZone(_this.whiteboardZone)) {
                 Entities.deleteEntity(_this.entityID);
             }
             _this.stopDrawing();
             if (laser) {
                 Entities.deleteEntity(laser);
                 laser = null;
             }
-            if (isTheTriggerUpdateRunning) {
+            if (_this.isTheTriggerUpdateRunning) {
                 Script.update.disconnect(_this.onTriggerPressedScriptUpdate);
-                isTheTriggerUpdateRunning = false;
+                _this.isTheTriggerUpdateRunning = false;
             }
         },
 
         /* On grip press, if the user is not in whiteboard zone, delete this sphere. If the user is using tablet or trigger 
         button, ignore. Begin a deleting interval that gets intersection data, updates laser, and deletes from the 
         current point. */
         gripPressed: function() {
-            if (!_this.isUserInZone(whiteboardZone)) {
+            if (!_this.isUserInZone(_this.whiteboardZone)) {
                 Entities.deleteEntity(_this.entityID);
             }
             if (tablet.tabletShown || activeTriggerPress) {
                 return;
             }
-            isTheGripUpdateRunning = true;
-            Script.update.connect(_this.onGripPressedScriptUpdate);
+            if (!_this.isTheGripUpdateRunning) {
+                Script.update.connect(_this.onGripPressedScriptUpdate);
```

Comment:
same deal - why can't we use a controller mapping? two handlers connected to the update loop in one script is a recipe for serious slowdown